### PR TITLE
Adds tests for Unix sockets

### PR DESCRIPTION
**Pull request by thanethomson**
As per #3115, adds simple Unix socket connect/accept deadline tests in
pretty much the same way as the TCP connect/accept deadline tests work.

* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: privval/socket_test.go [hunk 1]
```diff
@@ -82,3 +84,89 @@ func TestTCPListenerConnDeadline(t *testing.T) {
...
+	addr, err := testUnixAddr()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ln, err := net.Listen("unix", addr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	unixLn := NewUnixListener(ln)
```
Comment (melekes) [inline privval/socket_test.go]: both tcp and unix listeners should adhere to `net.Listener` interface. it means we can probably merge two tests into one (same thing we did with rpc clients https://github.com/tendermint/tendermint/blob/ec53ce359bb8f011e4dbb715da098bea08c32ded/rpc/client/rpc_test.go#L53). m?

Comment (thanethomson) [reply privval/socket_test.go]: I've combined the tests now into an iterable set of test cases. Does this work?

File: privval/socket_test.go [hunk 2]
```diff
@@ -18,67 +20,114 @@ func newPrivKey() ed25519.PrivKeyEd25519 {
...
+	UnixListenerConnDeadline(connectDeadline)(unixLn)
+	return listenerTestCase{
+		description: "Unix",
+		listener:    unixLn,
+		dialer:      DialUnixFn(addr),
+	}
+}
 
-		time.Sleep(2 * time.Millisecond)
+func constructListenerTestCases(t *testing.T, acceptDeadline, connectDeadline time.Duration) []listenerTestCase {
```
Comment (melekes) [inline privval/socket_test.go]:
```suggestion
func listenerTestCases(t *testing.T, acceptDeadline, connectDeadline time.Duration) []listenerTestCase {
```

I'd drop `construct` prefix

File: privval/socket_test.go (hunk 1 above)
Comment (melekes) [reply privval/socket_test.go]: Yep, thanks

**Pull request merged**